core/utils.py

```diff
@@ -0,0 +1,348 @@
+        frame = cv2.resize(
+            frame, (self._width, self._height), interpolation=cv2.INTER_AREA
+        )
+        if self._grayscale:
+            frame = np.expand_dims(frame, -1)
+
+        if self._key is None:
+            obs = frame
+        else:
+            obs = obs.copy()
+            obs[self._key] = frame
+        return obs
+
+
+def make_atari(env_id, skip=4, max_episode_steps=None):
+    """Make Atari games
+    Parameters
+    ----------
+    env_id: str
+        name of environment
+    skip: int
+        frame skip
+    max_episode_steps: int
+        max moves for an episode
+    """
+    env = gym.make(env_id)
+    assert 'NoFrameskip' in env.spec.id
+    env = NoopResetEnv(env, noop_max=30)
+    env = MaxAndSkipEnv(env, skip=skip)
+    if max_episode_steps is not None:
+        env = TimeLimit(env, max_episode_steps=max_episode_steps)
+    return env
+
+
+def set_seed(seed):
+    # set seed
+    random.seed(seed)
+    np.random.seed(seed)
+    torch.manual_seed(seed)
+    torch.cuda.manual_seed(seed)
+    torch.backends.cudnn.deterministic = True
+
+
+def make_results_dir(exp_path, args):
+    # make the result directory
+    os.makedirs(exp_path, exist_ok=True)
+    if args.opr == 'train' and os.path.exists(exp_path) and os.listdir(exp_path):
+        if not args.force:
+            raise FileExistsError('{} is not empty. Please use --force to overwrite it'.format(exp_path))
+        else:
+            print('Warning, path exists! Rewriting...')
+            shutil.rmtree(exp_path)
+            os.makedirs(exp_path)
+    log_path = os.path.join(exp_path, 'logs')
+    os.makedirs(log_path, exist_ok=True)
+    os.makedirs(os.path.join(exp_path, 'model'), exist_ok=True)
+    return exp_path, log_path
+
+
+def init_logger(base_path):
+    # initialize the logger
+    formatter = logging.Formatter('[%(asctime)s][%(name)s][%(levelname)s][%(filename)s>%(funcName)s] ==> %(message)s')
+    for mode in ['train', 'test', 'train_test', 'root']:
+        file_path = os.path.join(base_path, mode + '.log')
+        logger = logging.getLogger(mode)
+        handler = logging.StreamHandler()
+        handler.setFormatter(formatter)
+        logger.addHandler(handler)
+        handler = logging.FileHandler(file_path, mode='a')
+        handler.setFormatter(formatter)
+        logger.addHandler(handler)
+        logger.setLevel(logging.DEBUG)
+
+
+def select_action(visit_counts, temperature=1, deterministic=True):
+    """select action from the root visit counts.
+    Parameters
+    ----------
+    temperature: float
+        the temperature for the distribution
+    deterministic: bool
+        True -> select the argmax
+        False -> sample from the distribution
+    """
+    action_probs = [visit_count_i ** (1 / temperature) for visit_count_i in visit_counts]
+    total_count = sum(action_probs)
+    action_probs = [x / total_count for x in action_probs]
+    if deterministic:
```

Comment:
This deterministic. Where does it change and if it doesn't, can this be in our config?